install dependencies behind proxy

I'm **unable to install dependencies** when my network use **HTTP proxy**
```
pyb install_dependencies
```
> Going to execute task install_dependencies
> Installing all dependencies
> Installing build dependencies
> Installing dependency 'unittest-xml-reporting'
> 
> BUILD FAILED - Unable to install dependency 'unittest-xml-reporting'. See target/logs/install_dependencies\unittest-xml-reporting for details.

Extract of unittest-xml-reporting.err file :
> Retrying (Retry(total=0, connect=None, read=None, redirect=None)) after connection broken by 'ConnectTimeoutError(<pip._vendor.requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x033A16D0>, 'Connection to pypi.python.org timed out. (connect timeout=15)')': /simple/unittest-xml-reporting/

Environnement variable is set like below :
```
HTTP_PROXY=http://<proxy_ip_address>:<proxy_port>
```
I try this command line too :
```
pyb -E HTTP_PROXY=%HTTP_PROXY% install_dependencies
```
I think the **pip_utils.py** file must be updated to handle PROXY server.

A package could be installed behing proxy by using the command line below :
```
pip install --proxy=http://<proxy_ip_address>:<proxy_port> <package_name>
```

--- a/src/main/python/pybuilder/pip_utils.py
+++ b/src/main/python/pybuilder/pip_utils.py
@@ -17,6 +17,7 @@
 #   limitations under the License.
 
 import re
+import os
 from sys import version_info
 
 from pip._vendor.packaging.specifiers import SpecifierSet, InvalidSpecifier
@@ -72,6 +73,9 @@ def pip_install(install_targets, index_url=None, extra_index_url=None, upgrade=F
                                                       ))
     for install_target in as_list(install_targets):
         pip_command_line.extend(as_pip_install_target(install_target))
+
+    if env is None:
+        env = os.environ
 
     if logger:
         logger.debug("Invoking pip: %s", pip_command_line)

--- a/src/unittest/python/pip_utils_tests.py
+++ b/src/unittest/python/pip_utils_tests.py
@@ -16,7 +16,10 @@
 #   See the License for the specific language governing permissions and
 #   limitations under the License.
 
+import os
 import unittest
+
+from mock import patch, ANY
 
 from pybuilder import core
 from pybuilder import pip_utils
@@ -83,3 +86,18 @@ class PipVersionTests(unittest.TestCase):
             "--allow-unverified", "bar",
             "--allow-external", "bar"
         ])
+
+
+class PipUtilsTests(unittest.TestCase):
+    @patch("pybuilder.pip_utils.execute_command")
+    def test_pip_install_environ_inherited(self, execute_command):
+        pip_utils.pip_install("blah")
+        execute_command.assert_called_once_with(ANY, cwd=None, env=os.environ, error_file_name=None, outfile_name=None,
+                                                shell=False)
+
+    @patch("pybuilder.pip_utils.execute_command")
+    def test_pip_install_environ_overwritten(self, execute_command):
+        env_dict = dict()
+        pip_utils.pip_install("blah", env=env_dict)
+        execute_command.assert_called_once_with(ANY, cwd=None, env=env_dict, error_file_name=None, outfile_name=None,
+                                                shell=False)

